src/test/java/io/confluent/connect/elasticsearch/integration/BaseConnectorIT.java

```diff
@@ -0,0 +1,159 @@
+        CONNECTOR_STARTUP_DURATION_MS,
+        "Connector tasks did not start in time."
+    );
+    return System.currentTimeMillis();
+  }
+
+  /**
+   * Confirm that a connector with an exact number of tasks is running.
+   *
+   * @param connectorName the connector
+   * @param numTasks the minimum number of tasks
+   * @return true if the connector and tasks are in RUNNING state; false otherwise
+   */
+  protected Optional<Boolean> assertConnectorAndTasksRunning(String connectorName, int numTasks) {
+    try {
+      ConnectorStateInfo info = connect.connectorStatus(connectorName);
+      boolean result = info != null
+                       && info.tasks().size() >= numTasks
+                       && info.connector().state().equals(AbstractStatus.State.RUNNING.toString())
+                       && info.tasks().stream().allMatch(s -> s.state().equals(AbstractStatus.State.RUNNING.toString()));
+      return Optional.of(result);
+    } catch (Exception e) {
+      log.error("Could not check connector state info.", e);
+      return Optional.empty();
+    }
+  }
+
+  protected long waitConnectorToWriteDataIntoElasticsearch(String connectorName, int numTasks, String index, int numberOfRecords) throws InterruptedException {
+    TestUtils.waitForCondition(
+        () -> assertRecordsCountInElasticsearch(connectorName, numTasks, index, numberOfRecords).orElse(false),
+        CONSUME_MAX_DURATION_MS,
+        "Either writing into table has not started or row count did not matched."
+    );
+    return System.currentTimeMillis();
+  }
+
+  private Optional<Boolean> assertRecordsCountInElasticsearch(String name, int numTasks, String index, int numberOfRecords) {
+    try {
+      final JsonObject searchResult = client.search("", index, null);
+      final int recordsInserted = searchResult.getAsJsonObject("hits").getAsJsonObject("total").get("value").getAsInt();
+      ConnectorStateInfo info = connect.connectorStatus(name);
+      boolean result = info != null
+          && info.tasks().size() == numTasks
+          && info.connector().state().equals(AbstractStatus.State.RUNNING.toString())
+          && info.tasks().stream().allMatch(s -> s.state().equals(AbstractStatus.State.RUNNING.toString()));
+      boolean targetRowCountStatus = recordsInserted == numberOfRecords;
+      if (!targetRowCountStatus) {
+        log.warn("Row count did not matched. Expected {}, Actual {}.", numberOfRecords, recordsInserted);
+      }
+      return Optional.of(result && targetRowCountStatus);
+    } catch (Exception e) {
+      log.warn("Could not check connector state info, will retry shortly ...");
+      return Optional.empty();
+    }
+  }
+
+  protected void verifySearchResults(ConsumerRecords<byte[], byte[]> records, String index) throws IOException {
+    final JsonObject result = client.search("", index, null);
+    final int recordsInserted =result.getAsJsonObject("hits").getAsJsonObject("total").get("value").getAsInt();
```

Comment:
Any way to assert the contents?

Reply:
Looking for a possible way to assert the content.

Reply:
Added assert with actual data